src/modal.ts

```diff
@@ -189,89 +196,115 @@ export default class Modal extends LitElement {
-            </glide-core-modal-icon-button>
+      <div class="container" ${ref(this.#containerElementRef)}>
+        <header
+          class=${classMap({ header: true, scrolled: this.isScrolledFromTop })}
+        >
+          <div class="label-and-actions">
+            <h2 class="label" data-test="heading" id="heading">
+              ${when(
+                this.severity,
+                () =>
+                  html`<span
+                    aria-label="${this.#localize.term(
+                      this.severity === 'informational'
+                        ? 'severityInformational'
+                        : this.severity === 'critical'
+                          ? 'severityCritical'
+                          : 'severityMedium',
+                    )} - "
+                    class=${classMap({
+                      severity: true,
+                      critical: this.severity === 'critical',
+                      informational: this.severity === 'informational',
+                      medium: this.severity === 'medium',
+                    })}
+                    data-test="severity"
+                  >
+                    ${this.severity && icons[this.severity]}
+                  </span>`,
+              )}
+              ${when(
+                this.backButton && !this.severity,
+                () =>
+                  html`<glide-core-modal-icon-button
+                    class="back-button"
+                    data-test="back-button"
+                    label=${this.#localize.term('close')}
+                    @click=${this.#onCloseButtonClick}
+                    ${ref(this.#backButtonElementRef)}
+                  >
+                    ${icons.back}
+                  </glide-core-modal-icon-button>`,
+              )}
+              ${this.label}
+            </h2>
+
+            <div class="header-actions" role="toolbar">
+              <slot
+                name="header-actions"
+                ${assertSlot([ModalIconButton], true)}
+                ${ref(this.#headerActionsSlotElementRef)}
+              >
+                <!-- @type {ModalIconButton} -->
+              </slot>
+
+              <glide-core-modal-icon-button
+                class="close-button"
+                data-test="close-button"
+                label=${this.#localize.term('close')}
+                @click=${this.#onCloseButtonClick}
+                ${ref(this.#closeButtonElementRef)}
+              >
+                ${xIcon}
+              </glide-core-modal-icon-button>
+            </div>
           </div>
+
+          ${when(
+            this.description,
+            () =>
+              html`<h3
+                class="description"
+                data-test="description"
+                id="description"
+              >
+                ${this.description}
+              </h3>`,
+          )}
         </header>
 
-        <article aria-labelledby="heading" class="body" role="region">
+        <article
```

Comment:
I know `<article>` is existing. But does it add anything over a `<div>` given you're overriding its implicit "article" role?